connmgr/connmanager.go

```diff
@@ -463,14 +470,15 @@ func (cm *ConnManager) Connect(c *ConnReq) {
 
 // Disconnect disconnects the connection corresponding to the given connection
 // id. If permanent, the connection will be retried with an increasing backoff
-// duration.
-func (cm *ConnManager) Disconnect(id uint64) {
+// duration. If the reconnect bool is passed, we will attempt to reconnect
+// after the disconnect.
+func (cm *ConnManager) Disconnect(id uint64, reconnect bool) {
```

Comment:
Is this needed? I think we just actually need to call `Remove`, then we can manually queue the reconnection attempt ourselves.